Extension/src/LanguageServer/configurations.ts

```diff
@@ -441,7 +441,10 @@ export class CppProperties {
         // Valid compiler + IntelliSenseMode combinations:
         // 1. compiler is cl/clang-cl and IntelliSenseMode is MSVC
         // 2. compiler is not cl/clang-cl and IntelliSenseMode is not MSVC
-        const isValid: boolean = compilerPathAndArgs.compilerName.endsWith("cl.exe") === configuration.intelliSenseMode.startsWith("msvc");
+        const isValid: boolean = compilerPathAndArgs.compilerName.endsWith("cl.exe") ===
+            (configuration.intelliSenseMode.startsWith("msvc")
+            || configuration.intelliSenseMode.startsWith("windows-msvc")
+            || configuration.intelliSenseMode.startsWith("windows-clang"));
```

Comment:
I think the addition of "windows-clang" here is wrong.  1. On Windows, it's possible to use either clang or clang-cl.  2. We moved away from using MSVC intellisense mode for clang/clang-cl, and are instead using the clang intellisense mode with the appropriate ms extensions disabled.  This was addressed with https://github.com/microsoft/vscode-cpptools/issues/6075 (And changed by native PR: 274593)  I don't believe we should be matching up clang/clang-cl with MSVC intellisense mode anymore.  If you see any checks for this, they may be erroneous.

Reply:
Doh.  I misunderstood.  Disregard my last comment.

Was the intention to no longer squiggle use of clang intellisense mode when using "clang-cl.exe"?  If so, maybe this could be slightly refactored to compare the whole filename part == "cl.exe" rather than using endsWith(), and only check against MSVC modes?

Reply:
>Was the intention to no longer squiggle use of clang intellisense mode when using "clang-cl.exe"? 
Yeah, the use of `windows-clang*` mode indicates the use of "clang-cl.exe" for windows platform.